### PR TITLE
feat: form viewer

### LINKED ISSUE #177: Form Viewer component for response creation
# Overview 
Form-Viewer
Given some form schema, create a component to render the form

# Requirement

- [ ] Take in form schema, render all specified titles and fields

# Follow up
Create ticket to handle saving a response on the form

### PR DESCRIPTION
**Pull request by AnthonyLaiRH**
### Related Issue(s)

resolves #177 

### Quick overview of the change
Form Viewer which takes in form object from mongodb and renders the visual representation for the form

Included some basic db seeding for forms for testing purposes

Built test page for form viewer

### Quick testing guide

1. seed the mongoDB
2. `docker-compose up -d` on root dir
3. `docker-compose --build db_seed`
4. after seeing a successful seed, go to [`localhost:5555`](http://localhost:5555)
5. in primsa studio, check for seeded Form and copy the `id`
6. remaining in the Form, click on the `lastestFormVersion` and attach the only seeded FormVersion by clicking the checkbox
7. save changes in prisma studio
8. Now in terminal, `cd client`
9. `yarn dev`
10. in `src/pages/forms/ViewForm.tsx`, paste in the Form `id` copied from before
11. go to [`localhost:3000/forms/testViewForm`](http://localhost:3000/forms/testviewform)

### CODE REVIEW

File: client/src/pages/forms/ViewForm.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { FormViewer } from '../../components/form-viewer/form-viewer';
+import { useGetForm } from '../../data/form';
+
+const ViewForm = () => {
+  const { data } = useGetForm('6268d8c1a34ccc522cc5df19');
```
Comment (AnthonyLaiRH) [inline client/src/pages/forms/ViewForm.tsx:6]:
```suggestion
  const formId = '' // Add testing form id here
  const { data } = useGetForm(`${formId}`);
```

**Review by AnthonyLaiRH — COMMENTED** [review]
Author comments

File: client/src/components/form-viewer/form-viewer.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        <PageHeader pageTitle={form.title} backURL="/forms" />
+        <Spacer size="lg" />
+        {sections.map((section, i) => (
+          <>
+            <Card key={section.id}>
+              <CardContent>
+                <Typography variant="h4" sx={{ mb: 2 }}>
+                  Section #{i + 1} - {section.name}
+                </Typography>
+                <Typography variant="h5" sx={{ mb: 2 }}>
```
Comment (callmekungfu) [inline client/src/components/form-viewer/form-viewer.tsx:32]: I would use `body` or `body2` instead for description

File: client/src/components/form-viewer/view-form-field.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  const { control, setValue } = useFormContext();
+
+  switch (type) {
+    case 'text':
+      return (
+        <FormInputField
+          name={name}
+          control={control}
+          rules={{ required: !optional }}
+          label={description || name}
```
Comment (callmekungfu) [inline client/src/components/form-viewer/view-form-field.tsx:41]: The form label should just be the name itself. The label should never be the description.

We can enhance these field components to support help text which will allow us to render this information. I can log that as a follow up.

Also the preferred syntax for coalesce is `??` and not `||`

**Review by callmekungfu — COMMENTED** [review]
looks good on the first pass, have some concerns regarding the complex switch case system that I am seeing. Would prefer to switch to a more broken down record style approach instead.

File: client/src/components/form-viewer/form-viewer.tsx (hunk 2 above)
Comment (callmekungfu) [reply client/src/components/form-viewer/form-viewer.tsx:32]: Also the component should only be rendered if there is a need to render it

**Review by callmekungfu — APPROVED** [review]
Lgtm!

**Pull request merged**